src/SegmentedMsg.chpl

```diff
@@ -987,45 +880,81 @@ module SegmentedMsg {
     return chplIdx;
   }
 
-  proc segSliceIndex(objtype: string, objName: string, key: [] string,
+  proc segSliceIndex(objtype: string, objName: string, key: [] string, dtype: DType,
                                          st: borrowed SymTab): MsgTuple throws {
     var pn = Reflection.getRoutineName();
+    var repMsg: string;
 
     // check to make sure symbols defined
     st.checkTable(objName);
 
+    // Parse the slice parameters
+    var start = key[0]:int;
+    var stop = key[1]:int;
+    var stride = key[2]:int;
+
+    // Only stride-1 slices are allowed for now
+    if (stride != 1) { 
+        var errorMsg = notImplementedError(pn, "stride != 1"); 
+        smLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);      
+        return new MsgTuple(errorMsg, MsgType.ERROR);
+    }
+
+    // TO DO: in the future, we will force the client to handle this
+    var slice = convertPythonSliceToChapel(start, stop);
+
     select objtype {
         when "str" {
             // Make a temporary string array
             var strings = getSegString(objName, st);
 
-            // Parse the slice parameters
-            var start = key[0]:int;
-            var stop = key[1]:int;
-            var stride = key[2]:int;
-
-            // Only stride-1 slices are allowed for now
-            if (stride != 1) { 
-                var errorMsg = notImplementedError(pn, "stride != 1"); 
-                smLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);      
-                return new MsgTuple(errorMsg, MsgType.ERROR);
-            }
-            // TO DO: in the future, we will force the client to handle this
-            var slice = convertPythonSliceToChapel(start, stop);
             // Compute the slice
             var (newSegs, newVals) = strings[slice];
             // Store the resulting offsets and bytes arrays
             var newStringsObj = getSegString(newSegs, newVals, st);
-            var repMsg = "created " + st.attrib(newStringsObj.name) + "+created bytes.size %t".format(newStringsObj.nBytes);
-            smLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg); 
-            return new MsgTuple(repMsg, MsgType.NORMAL);
+            repMsg = "created " + st.attrib(newStringsObj.name) + "+created bytes.size %t".format(newStringsObj.nBytes);
+        }
+        when "SegArray" {
+          var rtnmap: map(string, string);
```

Comment:
i dont think `rtnmap` is used here

Reply:
I missed the update to use it. It is now